core/dbt/cli/resolvers.py

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+
+
+def default_project_dir():
+    cwd = Path.cwd()
+    root_path = Path(cwd.root)
+
+    while cwd != root_path:
+        # Use the directory if there is a profiles.yml file
+        if (cwd / "dbt_project.yml").exists():
+            return Path(cwd)
+        cwd = cwd.parent
+
+    return Path.cwd()
+
+
+def default_profiles_dir():
+    default_profiles_dir = Path.home() / ".dbt"
```

Comment:
This looks great, but we can tighten it up a bit by just using a conditional return here.
```
def default_profiles_dir():
    return Path.cwd() if (Path.cwd() / "profiles.yml").exists() else Path.home() / ".dbt"
````